### PR TITLE
Tutorial translation

### PR DESCRIPTION
**Pull request by balazsorban44**
Translating the Tutorial.

#1

### CODE REVIEW

**Comment by balazsorban44** [comment]
I request both @gergely-nagy and @punkah for reviewing this, as this is by far the longest translation, and even though I tried fixing many of my mistakes, there might be some uncaught ones still.

**Comment by balazsorban44** [comment]
ping @gergely-nagy, @punkah would it be realistic to get this reviewed this weekend? Now, that we got our domain, and we are so close to the finishing of core translations, I hope more and more people will find us. The tutorial section is something that would probably have a huge benefit for beginners!

File: content/tutorial/nav.yml [hunk 1]
```diff
@@ -1,111 +1,111 @@
...
           href: /tutorial/tutorial.html#what-is-react
           forceInternal: true
         - id: inspecting-the-starter-code
-          title: Inspecting the Starter Code
+          title: Kezdő kód ellenőrzése
           href: /tutorial/tutorial.html#inspecting-the-starter-code
           forceInternal: true
         - id: passing-data-through-props
-          title: Passing Data Through Props
+          title: Adattovábbítás prop-okkal
```
Comment (punkah) [inline content/tutorial/nav.yml]:
```suggestion
          title: Adattovábbítás propokkal
```

File: content/tutorial/tutorial.md [hunk 2]
```diff
@@ -12,130 +12,130 @@ redirect_from:
...
 ---
 
-This tutorial doesn't assume any existing React knowledge.
+Ez a tutoriál nem feltételez korábbi React ismereteket.
 
-## Before We Start the Tutorial {#before-we-start-the-tutorial}
+## Mielőtt elkezdjük a Tutoriált {#before-we-start-the-tutorial}
 
-We will build a small game during this tutorial. **You might be tempted to skip it because you're not building games -- but give it a chance.** The techniques you'll learn in the tutorial are fundamental to building any React app, and mastering it will give you a deep understanding of React.
+Ebben a tutoriálban egy kis játékot fogunk készíteni. **Csábító lehet átugrani, mivel nem játékokat készítesz -- de azért adj neki egy esélyt.** A technikák amiket itt tanulsz alapvetőek bármilyen React alkalmazáshoz, és ha ezeket sikerül elsajátítanod, úgy sokkal jobban meg fogod érteni a React működését.
```
Comment (punkah) [inline content/tutorial/tutorial.md]:
```suggestion
Ebben a tutoriálban egy kis játékot fogunk készíteni. **Csábító lehet átugrani, mivel nem játékokat készítesz -- de azért adj neki egy esélyt.** A technikák, amiket itt tanulsz, alapvetőek bármilyen React alkalmazáshoz, és ha ezeket sikerül elsajátítanod, úgy sokkal jobban meg fogod érteni a React működését.
```

File: content/tutorial/tutorial.md [hunk 3]
```diff
@@ -12,130 +12,130 @@ redirect_from:
...
 
-## Setup for the Tutorial {#setup-for-the-tutorial}
+## Beállítások a Tutoriálhoz {#setup-for-the-tutorial}
 
-There are two ways to complete this tutorial: you can either write the code in your browser, or you can set up a local development environment on your computer.
+Ezt a tutoriált kétféleképpen is elvégezheted: kódolhatsz a böngésződből, vagy felállíthatsz egy helyi fejlesztői környezetet.
 
-### Setup Option 1: Write Code in the Browser {#setup-option-1-write-code-in-the-browser}
+### "1. Opció: kódolj a böngésződben" {#setup-option-1-write-code-in-the-browser}
```
Comment (punkah) [inline content/tutorial/tutorial.md]:
```suggestion
### 1. Opció: kódolj a böngésződben {#setup-option-1-write-code-in-the-browser}
```

File: content/tutorial/tutorial.md [hunk 4]
```diff
@@ -12,130 +12,130 @@ redirect_from:
...
+A leggyorsabban így állhatsz neki!
 
-First, open this **[Starter Code](https://codepen.io/gaearon/pen/oWWQNa?editors=0010)** in a new tab. The new tab should display an empty tic-tac-toe game board and React code. We will be editing the React code in this tutorial.
+Először is nyisd meg a **[Kezdő kódot](https://codepen.io/gaearon/pen/oWWQNa?editors=0010)** egy új fülön. Az új fül egy új, üres tic-tac-toe táblát, és a React kódot kell hogy mutassa. Ez a React az, amit ebben a tutoriálban szerkeszteni fogunk.
 
-You can now skip the second setup option, and go to the [Overview](#overview) section to get an overview of React.
+Ugord át a második opciót, és a React áttekintéséhez menj az [Áttekintés](#overview) szekcióhoz.
 
-### Setup Option 2: Local Development Environment {#setup-option-2-local-development-environment}
+### "2. Opció: helyi fejlesztői környezet" {#setup-option-2-local-development-environment}
```
Comment (punkah) [inline content/tutorial/tutorial.md]:
```suggestion
### 2. Opció: helyi fejlesztői környezet {#setup-option-2-local-development-environment}
```

File: content/tutorial/tutorial.md [hunk 5]
```diff
@@ -12,130 +12,130 @@ redirect_from:
...
+Ezt a tutoriált kétféleképpen is elvégezheted: kódolhatsz a böngésződből, vagy felállíthatsz egy helyi fejlesztői környezetet.
 
-### Setup Option 1: Write Code in the Browser {#setup-option-1-write-code-in-the-browser}
+### "1. Opció: kódolj a böngésződben" {#setup-option-1-write-code-in-the-browser}
 
-This is the quickest way to get started!
+A leggyorsabban így állhatsz neki!
 
-First, open this **[Starter Code](https://codepen.io/gaearon/pen/oWWQNa?editors=0010)** in a new tab. The new tab should display an empty tic-tac-toe game board and React code. We will be editing the React code in this tutorial.
+Először is nyisd meg a **[Kezdő kódot](https://codepen.io/gaearon/pen/oWWQNa?editors=0010)** egy új fülön. Az új fül egy új, üres tic-tac-toe táblát, és a React kódot kell hogy mutassa. Ez a React az, amit ebben a tutoriálban szerkeszteni fogunk.
```
Comment (punkah) [inline content/tutorial/tutorial.md]:
```suggestion
Először is nyisd meg a **[Kezdő kódot](https://codepen.io/gaearon/pen/oWWQNa?editors=0010)** egy új fülön. Az új fül egy új, üres tic-tac-toe táblát és a React kódot kell, hogy mutassa. Ez a React kód az, amit ebben a tutoriálban szerkeszteni fogunk.
```

File: content/tutorial/tutorial.md [hunk 6]
```diff
@@ -12,130 +12,130 @@ redirect_from:
...
+### Segítség, elakadtam! {#help-im-stuck}
 
-If you get stuck, check out the [community support resources](/community/support.html). In particular, [Reactiflux Chat](https://discord.gg/reactiflux) is a great way to get help quickly. If you don't receive an answer, or if you remain stuck, please file an issue, and we'll help you out.
+Ha bármikor elakadsz, a [közösségi támogatási források](/community/support.html) segíthet. Különösen a [Reactiflux Chat](https://discord.gg/reactiflux) lehet hasznos, ha gyorsan szeretnél segítséget kapni. Ha nem érkezik válasz, vagy még mindig el vagy akadva, nyiss egy issue-t és segítünk.
 
-## Overview {#overview}
+## Áttekintés {#overview}
 
-Now that you're set up, let's get an overview of React!
+Most hogy minden készen áll, kezdjük a React áttekintésével!
```
Comment (punkah) [inline content/tutorial/tutorial.md]:
```suggestion
Most, hogy minden készen áll, kezdjük a React áttekintésével!
```

File: content/tutorial/tutorial.md [hunk 7]
```diff
@@ -12,130 +12,130 @@ redirect_from:
...
 
-### What Is React? {#what-is-react}
+### Mi az a React? {#what-is-react}
 
-React is a declarative, efficient, and flexible JavaScript library for building user interfaces. It lets you compose complex UIs from small and isolated pieces of code called "components".
+A React egy deklaratív, effektív, és rugalmas JavaScript könyvtár, felhasználói felületek készítéséhez. Lehetővé teszi komplex felhasználói felületek összeállítását izolált kódrészletekből, amiket "komponenseknek" hívunk.
 
-React has a few different kinds of components, but we'll start with `React.Component` subclasses:
+A React rendelkezik egy pár komponenstípussal, de most kezdjük a `React.Comoponent` alosztállyal:
```
Comment (punkah) [inline content/tutorial/tutorial.md]:
```suggestion
A React rendelkezik egy pár komponenstípussal, de most kezdjük a `React.Component` alosztállyal:
```

File: content/tutorial/tutorial.md [hunk 8]
```diff
@@ -146,51 +146,51 @@ class ShoppingList extends React.Component {
   }
 }
 
-// Example usage: <ShoppingList name="Mark" />
+// Példa használata: <ShoppingList name="Mark" />
 ```
 
-We'll get to the funny XML-like tags soon. We use components to tell React what we want to see on the screen. When our data changes, React will efficiently update and re-render our components.
+Nemsokára beszélünk a vicces XML szerű címkékről is. A komponensek segítségével mondjuk meg a Reactnek, hogy mit szeretnénk látni a képernyőn. Ha az adatunk megváltozik, a React hatékonyan frissíti és újrarendereli a komponensünket.
```
Comment (punkah) [inline content/tutorial/tutorial.md]:
```suggestion
Nemsokára beszélünk a vicces XML-szerű címkékről is. A komponensek segítségével mondjuk meg a Reactnek, hogy mit szeretnénk látni a képernyőn. Ha az adatunk megváltozik, a React hatékonyan frissíti és újrarendereli a komponensünket.
```

File: content/tutorial/tutorial.md [hunk 9]
```diff
@@ -146,51 +146,51 @@ class ShoppingList extends React.Component {
...
 
-The `render` method returns a *description* of what you want to see on the screen. React takes the description and displays the result. In particular, `render` returns a **React element**, which is a lightweight description of what to render. Most React developers use a special syntax called "JSX" which makes these structures easier to write. The `<div />` syntax is transformed at build time to `React.createElement('div')`. The example above is equivalent to:
+A `render` metódus egy *leírását* adja vissza annak, amit a képernyőn szeretnél látni. A React fogja a leírást, és megjeleníti az eredményt. Pontosabban, a `render` metódus egy **React elem**-et ad vissza, ami egy könnyűsúlyú leírása annak, amit renderelni kell. A legtöbb React fejlesztő egy speciális szintaxist használ, ezt "JSX"-nek hívják, ami könnyebbé teszi ezen a struktúrák írását. A `<div />` szintaxist `React.createEelement('div')`-té transzformáljuk kompiláláskor. A fenti példa egyenértékű az alábbival:
```
Comment (punkah) [inline content/tutorial/tutorial.md]:
```suggestion
A `render` metódus egy *leírását* adja vissza annak, amit a képernyőn szeretnél látni. A React fogja a leírást, és megjeleníti az eredményt. Pontosabban, a `render` metódus egy **React elem**-et ad vissza, ami egy könnyűsúlyú leírása annak, amit renderelni kell. A legtöbb React fejlesztő egy speciális szintaxist használ, ezt "JSX"-nek hívják, ami könnyebbé teszi ezen struktúrák írását. A `<div />` szintaxist `React.createEelement('div')`-té transzformáljuk kompiláláskor. A fenti példa egyenértékű az alábbival:
```

File: content/tutorial/tutorial.md [hunk 10]
```diff
@@ -1010,8 +1011,8 @@ Let's `map` over the `history` in the Game's `render` method:
 
     const moves = history.map((step, move) => {
       const desc = move ?
-        'Go to move #' + move :
-        'Go to game start';
+        'Menj erre ide, lépés: #' + move :
```
Comment (punkah) [inline content/tutorial/tutorial.md]:
```suggestion
        'Menj ide, lépés: #' + move :
```

File: content/tutorial/tutorial.md [hunk 11]
```diff
@@ -1043,62 +1044,62 @@ Let's `map` over the `history` in the Game's `render` method:
...
+**[Nézd meg a teljes kódot ezen a ponton](https://codepen.io/gaearon/pen/EmmGEa?editors=0010)**
 
-For each move in the tic-tac-toes's game's history, we create a list item `<li>` which contains a button `<button>`. The button has a `onClick` handler which calls a method called `this.jumpTo()`. We haven't implemented the `jumpTo()` method yet. For now, we should see a list of the moves that have occurred in the game and a warning in the developer tools console that says:
+A tic-tac-toe játék lépéstörténetében minden lépéshez létrehozunk egy `<li>`-t, ami tartalmaz egy `<button>` gombot. A gomb rendelkezik egy `onClick` kezelővel, ami meghív egy `this.jumpTo()` metódust. A `jumpTo()` metódust még nem implementáltuk. Egyenlőre egy listát kell látnunk a lépésekről, amik már megtörténtek a játék során, és egy figyelmeztetést a fejlesztői eszközözök konzolban, ami azt mondja:
 
 >  Warning:
 >  Each child in an array or iterator should have a unique "key" prop. Check the render method of "Game".
 
-Let's discuss what the above warning means.
+Beszéljük meg mit is jelent a fenti figyelmeztetés.
```
Comment (punkah) [inline content/tutorial/tutorial.md]:
```suggestion
Beszéljük meg, mit is jelent a fenti figyelmeztetés.
```

**Comment by balazsorban44** [comment]
@gergely-nagy Could you have a look at this, if you have time?

**Pull request merged**